hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogStateManagerImpl.java

```diff
@@ -209,6 +209,23 @@ public void increaseRetryCountOfTransactionInDB(
     }
   }
 
+  @Override
+  public int resetRetryCountOfTransactionInDB(ArrayList<Long> txIDs)
+      throws IOException {
+    for (long txId: txIDs) {
+      DeletedBlocksTransaction transaction = deletedTable.get(txId);
```

Comment:
> We need to handle the case where txn is not present in db.

This is already handled in `DeletedBlockLogImpl`,  only the filtered txns will be processed.


> Also what if retry count of this transaction is > 0 already?

In this case, we do nothing for them now.

Reply:
>This is already handled in DeletedBlockLogImpl, only the filtered txns will be processed.

I think we should still handle this case, o.w we may end up hitting NPE.

Reply:
> > This is already handled in DeletedBlockLogImpl, only the filtered txns will be processed.
> 
> I think we should still handle this case, o.w we may end up hitting NPE.

Do you mean adding a null check here?
```suggestion
      Objects.requireNonNull(txId, "transactionId cannot be null");
      DeletedBlocksTransaction transaction = deletedTable.get(txId);
```

Reply:
If retry count is > 0 for a transaction, we should not reset its count.
It is possible user tries to reset count for non-existent transactions or transactions already being processed.

Reply:
@lokeshj1703  Actually, this action is buffered and only takes effect(flushed) when SCM `take snapshot`, due to this uncontrollable flush interval, such a case is inevitable.
For example:
user send cmd for the first time
user find RDB not take effect
buffer flushes
txn retry count increase from 0 to be 1 ( increment count action also will be added into this buffer)
user send cmd for the second time
buffer flushes

For the whole process, txn retry count is  -1 -> 0 -> 1 -> 0 . Even if we add a check when getting this txn from DB and before adding the change into the buffer, this case still could happen.

Reply:
I have removed the manual flush according to the discussion with Sammi since SCM buffer needs an external regular check and flush as a common solution, not in this case as a certain solution.

Reply:
Ideally the scenario should not happen, we should keep the transactionBuffer data in memory as well I think.
But we can still make best effort anyways.